composites/Plugin/SnippetPreview/components/SnippetPreview.js

```diff
@@ -1,17 +1,625 @@
+			activeField,
+		} = this.props;
+
+		if ( activeField === fieldName ) {
+			return addCaretStyle( BaseComponent, colorCaret, mode );
+		}
+
+		if ( hoveredField === fieldName ) {
+			return addCaretStyle( BaseComponent, colorCaretHover, mode );
+		}
+
+		return BaseComponent;
+	}
+
+	/**
+	 * Returns the breadcrumbs string to be rendered.
+	 *
+	 * @returns {string} The breadcrumbs.
+	 */
+	getBreadcrumbs() {
+		const { url, breadcrumbs } = this.props;
+		const { protocol, hostname, pathname } = parse( url );
+
+		const hostPart = protocol === "https:" ? protocol + "//" + hostname : hostname;
+
+		const urlParts = breadcrumbs || pathname.split( "/" );
+
+		return [ hostPart, ...urlParts ].filter( part => !! part ).join( " › " );
+	}
+
+	/**
+	 * Renders the URL for display in the snippet preview.
+	 *
+	 * @returns {ReactElement} The rendered URL.
+	 */
+	renderUrl() {
+		const {
+			url,
+			onClick,
+			onMouseOver,
+			onMouseLeave,
+			locale,
+			keyword,
+		} = this.props;
+
+		let urlContent;
+
+		if ( this.props.mode === MOBILE ) {
+			urlContent = this.getBreadcrumbs();
+		} else {
+			urlContent = highlightKeyword( locale, keyword, url );
+		}
+
+		const Url = this.addCaretStyles( "url", BaseUrl );
+
+		return <Url onClick={ onClick.bind( null, "url" ) }
+		            onMouseOver={ partial( onMouseOver, "url" ) }
+		            onMouseLeave={ partial( onMouseLeave, "url" ) }>
+			{ urlContent }
+		</Url>;
+	}
+
+	/**
+	 * Before we receive props we need to set the title and description in the
+	 * state.
+	 *
+	 * @param {Object} nextProps The props this component will receive.
+	 *
+	 * @returns {void}
+	 */
+	componentWillReceiveProps( nextProps ) {
+		const nextState = {};
+
+		if ( this.props.title !== nextProps.title ) {
+			nextState.title = nextProps.title;
+		}
+
+		if ( this.props.description !== nextProps.description ) {
+			nextState.description = nextProps.description;
+		}
+
+		this.setState( nextState );
+	}
+
+	/**
+	 * After a component updates we need to fit the title.
+	 *
+	 * @returns {void}
+	 */
+	componentDidUpdate() {
+		if ( this.props.mode === MOBILE ) {
+			this.fitTitle();
+			this.fitDescription();
+		}
+	}
+
+	/**
+	 * Renders the snippet preview.
+	 *
+	 * @returns {ReactElement} The rendered snippet preview.
+	 */
+	render() {
+		const {
+			keyword,
+			isDescriptionGenerated,
+			locale,
+			onClick,
+			onMouseLeave,
+			onMouseOver,
+			mode,
+			isAmp,
+		} = this.props;
+
+		const {
+			PartContainer,
+			Container,
+			TitleUnbounded,
+			Title,
+			Description,
+		} = this.getPreparedComponents( mode );
+
+		const separator = mode === DESKTOP ? null : <Separator/>;
+		const downArrow = mode === DESKTOP ? <UrlDownArrow/> : null;
+		const amp       = mode === DESKTOP || ! isAmp ? null : <Amp/>;
+
+		const renderedDate = this.renderDate();
+
+		return (
+			<section>
+				<Container onMouseLeave={this.onMouseLeave} width={ MAX_WIDTH + 2 * WIDTH_PADDING } padding={ WIDTH_PADDING }>
```

Comment:
If you use the codestyle someProp={`space`someVariable`space`} here, then you are missing some in the next section.